### PR TITLE
MRG: group_by to evoked.plot_image

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
And the last one for `evoked.plot_image` from #5170

Adds a group_by option to evoked.plot_image that allows feeding a dict to `axes`. I.e., for making a channel selection with `make_1020_selection`.

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
The whatsnew entries and python_reference entries are here @agramfort

**Comment by jona-sassenhagen** [comment]
With this as the last of the 3 patches, it is very easy to make plots like this:

unknown-141

```
rois = make_1020_channel_selections(evoked.info, midline="12z")

fig, axes = plt.subplots(nrows=3)
axes = {roi: ax for roi, ax in zip(rois, axes.flatten())}
evoked.plot_image(group_by=rois, axes=axes, show=False, show_names='all', colorbar=False)
plt.colorbar(axes["Left"].images[0], ax=list(axes.values()), shrink=.3)
```

**Comment by jona-sassenhagen** [comment]
https://8265-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_stats_cluster_erp.html#sphx-glr-auto-tutorials-plot-stats-cluster-erp-py

Example looks good.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -23,6 +23,8 @@ Changelog
 
 - Add ability to pass a precomputed forward solution to :func:`mne.simulation.simulate_raw` by `Eric Larson`_
 
+- Add :func:`mne.channels.make_1020_channel_selections` to group 10/20-named EEG channels by location, `Jona Sassenhagen`_
+
```
Comment (cbrnr) [inline doc/whats_new.rst]: by `Jona Sassenhagen`_

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -200,8 +200,62 @@ def _plot_evoked(evoked, picks, exclude, unit, show, ylim, proj, xlim, hline,
...
+    # Either both must be dicts, or neither.
+    # If the former, the two dicts provide picks and axes to plot them to.
+    # Then, we call this function recursively for each entry in `group_by`.
+    if plot_type == "image" and isinstance(group_by, dict):
+        if axes is None:
+            axes = dict()
+            for sel in group_by:
+                plt.figure()
+                axes[sel] = plt.axes()
+        elif isinstance(axes, dict):
```
Comment (cbrnr) [inline mne/viz/evoked.py]: Why not raise the error with `elif not isinstance(axes, dict)`?

File: mne/viz/evoked.py [hunk 3]
```diff
@@ -200,8 +200,62 @@ def _plot_evoked(evoked, picks, exclude, unit, show, ylim, proj, xlim, hline,
...
+        min, max = clims.min(), clims.max()
+        for im in ims:
+            im.set_clim(min, max)
+        figs = [ax.get_figure() for ax in axes.values()]
+        if len(set(figs)) is 1:
+            return figs[0]
+        else:
+            return figs
+    elif isinstance(axes, dict):
+        raise ValueError("If `group_by` isn't a dict, neither can `axes` be.")
```
Comment (cbrnr) [inline mne/viz/evoked.py]: If `group_by` is not a dict, `axes` must not be a dict either.

File: mne/viz/evoked.py [hunk 4]
```diff
@@ -836,10 +890,14 @@ def plot_evoked_image(evoked, picks=None, exclude='bads', unit=True,
     titles : dict | None
         The titles associated with the channels. If None, defaults to
         ``dict(eeg='EEG', grad='Gradiometers', mag='Magnetometers')``.
-    axes : instance of Axis | list | None
+    axes : instance of Axis | list | dict | None
         The axes to plot to. If list, the list must be a list of Axes of
         the same length as the number of channel types. If instance of
         Axes, there must be only one channel type plotted.
+        If ``group_by`` is a dict, this cannot be a list, but it can be a dict
```
Comment (cbrnr) [inline mne/viz/evoked.py]: I'd use single backticks to refer to other args.

File: mne/viz/evoked.py (hunk 2 above)
Comment (jona-sassenhagen) [reply mne/viz/evoked.py]: ...

**Comment by jona-sassenhagen** [comment]
Would be super cool if we could get this merged by the weekend :)

**Comment by agramfort** [comment]
ok to merge when Green

**Comment by jona-sassenhagen** [comment]
@cbrnr fine?

**Comment by cbrnr** [comment]
You wanna wait for @mmagnuski?

**Comment by jona-sassenhagen** [comment]
@mmagnuski would be super nice if you could provide some comments in the next hour or so ...

File: tutorials/plot_stats_cluster_erp.py [hunk 5]
```diff
@@ -117,21 +117,11 @@
...
-                      clim=dict(eeg=(-vmax, vmax)), mask=significant_points,
-                      **time_unit, show_names="all")
-    evoked.nave = None  # to suppress printing trial count for other images
-    if not ax.is_last_row():  # remove xticklabels for all but bottom axis
-        ax.set(xlabel='', xticklabels=[])
-    ax.set(ylabel='', title=selection_name)
-
-fig.colorbar(ax.images[-1], ax=axes, fraction=.1, aspect=20,
-             pad=.05, shrink=2 / 3, label="uV", orientation="vertical")
+axes = {sel: ax for sel, ax in zip(selections, axes.ravel().tolist())}
```
Comment (larsoner) [inline tutorials/plot_stats_cluster_erp.py]: you should not need `tolist()` here

Comment (jona-sassenhagen) [reply tutorials/plot_stats_cluster_erp.py]: Yup, still works.

**Comment by jona-sassenhagen** [comment]
Merge :)

**Comment by larsoner** [comment]
Thanks @jona-sassenhagen

**Comment by jona-sassenhagen** [comment]
Great, thanks!

**Comment by mmagnuski** [comment]
@jona-sassenhagen 
Sorry, I was away from github for some time and didn't check notifications.
I am still alive and will be back in about a week. :)

**Pull request merged**